File: utils/updates.go
package utils

import (
	"fmt"
	"os/exec"
)

func UpdateSystem(desktopEnv string) {
	var pkexecCmd *exec.Cmd

	switch desktopEnv {
	case "xfce":
		pkexecCmd = exec.Command("xfce4-terminal", "-x", "pkexec", "pacman", "--noconfirm", "-Syu")
		if err := pkexecCmd.Run(); err != nil {
			fmt.Printf("Error executing command: %v\n", err)
		}
	case "gnome":
		pkexecCmd = exec.Command("gnome-terminal", "--", "sudo", "pacman", "--noconfirm", "-Syu")
		if err := pkexecCmd.Run(); err != nil {
			fmt.Printf("Error executing command: %v\n", err)
		}
	case "kde":
		pkexecCmd = exec.Command("konsole", "-e", "sudo", "pacman", "--noconfirm", "-Syu")
		if err := pkexecCmd.Run(); err != nil {
			fmt.Printf("Error executing command: %v\n", err)
		}
	default:
		fmt.Printf("unsupported desktop environment: %s\n", desktopEnv)
		if err := pkexecCmd.Run(); err != nil {
			fmt.Printf("Error executing command: %v\n", err)
		}
	}
}

File: utils/updates.py
"""
System update utilities
"""

import subprocess
import os

def update_system(desktop_env):
    """
    Update the system using pacman in a terminal appropriate for the desktop environment
    Args:
        desktop_env: str - Desktop environment name (lowercase)
    """
    try:
        if desktop_env == "xfce":
            subprocess.Popen([
                'xfce4-terminal', '-x', 'pkexec', 'pacman', '--noconfirm', '-Syu'
            ])
        elif desktop_env == "gnome":
            subprocess.Popen([
                'gnome-terminal', '--', 'sudo', 'pacman', '--noconfirm', '-Syu'
            ])
        elif desktop_env == "kde":
            env = os.environ.copy()
            
            env.pop('LD_LIBRARY_PATH', None)
            env.pop('QT_PLUGIN_PATH', None)
            env.pop('QT_QPA_PLATFORM_THEME', None)
            subprocess.Popen([
                'konsole', '-e', 'sudo', 'pacman', '--noconfirm', '-Syu'
            ], env=env)
        else:
            print(f"Unsupported desktop environment: {desktop_env}")
    except Exception as e:
        print(f"Error executing command: {e}")